File: ExemplosR/Exemplo1.py
import random

# sample == random.choices
nsamples = 100000
Pb = 0
Pa = 0

for i in range(nsamples):
    face1 = random.choices(population=["CA", "CO"], weights=[2/5, 3/5], k=1)[0]
    face2 = random.choices(population=["CA", "CO"], weights=[2/5, 3/5], k=1)[0]
    face3 = random.choices(population=["CA", "CO"], weights=[2/5, 3/5], k=1)[0]
    
    if (face1 == "CA" and face2 == "CO") or (face1 == "CO" and face2 == "CA"):
        Pa += 1
        
    if (face2 == "CO" and face3 == "CO") or (face2 == "CO" and face3 == "CO"):
        Pb += 1
        
print("P(A): ", Pa/nsamples)
print("P(B): ", Pb/nsamples)

File: ExemplosR/Exemplo1.R
nsamples <- 100000
cont.A <- 0
cont.B <- 0

for (i in 1:nsamples) {
  
  face1 <- sample(c("CA","CO"), 1, prob = c(2/5, 3/5))
  face2 <- sample(c("CA","CO"), 1, prob = c(2/5, 3/5))
  face3 <- sample(c("CA","CO"), 1, prob = c(2/5, 3/5))
  
  if (((face1 == "CA") & (face2 == "CO")) | ((face1 == "CO") & (face2 == "CA"))) {
    cont.A <- cont.A + 1
  }
  
  if ((face2 == "CO") & (face3 == "CO")) {
    cont.B <- cont.B + 1
  }

}

PA <- cont.A / nsamples
print(PA)

PB <- cont.B / nsamples
print(PB)
